daemons/controld/controld_cib.c

```diff
@@ -437,9 +438,10 @@ controld_delete_resource_history(const char *rsc_id, const char *node,
 
     // Ask CIB to delete the entry
     xpath = crm_strdup_printf(XPATH_RESOURCE_HISTORY, node, rsc_id);
-    rc = cib_internal_op(controld_globals.cib_conn, PCMK__CIB_REQUEST_DELETE,
-                         NULL, xpath, NULL, NULL, call_options|cib_xpath,
-                         user_name);
+
+    cib->cmds->set_user(cib, user_name);
+    rc = cib->cmds->remove(cib, xpath, NULL, call_options|cib_xpath);
+    cib->cmds->set_user(cib, NULL);
```

Comment:
previously, a null connection would segfault in `cib_internal_op()`, so this isn't any worse, but if we're explicitly dereferencing it here, this would be a good time to guard it

Reply:
We've already returned by now if `cib == NULL`. But we're checking it as `controld_globals.cib_conn` when checking it as `cib` would be clearer. I'll update.